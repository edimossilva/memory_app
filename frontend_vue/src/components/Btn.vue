<template>
  <button @click="onClick" :dataId="dataId">{{ label }}</button>
</template>

<script>
export default {
  name: "Btn",
  props: {
    label: {
      type: String,
      required: true,
    },

    onClick: {
      type: Function,
      required: false,
      default: () => {},
    },

    dataId: String,
  },
};
</script>
