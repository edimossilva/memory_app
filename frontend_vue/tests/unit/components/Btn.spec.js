import { shallowMount } from "@vue/test-utils";
import Btn from "@/components/Btn.vue";
import Vue from "vue";
import VueRouter from "vue-router";

Vue.use(VueRouter);
describe("Btn.vue", () => {
  it("Matchs Snapshot", () => {
    const wrapper = shallowMount(Btn, {
      propsData: { label: "mock label" },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
