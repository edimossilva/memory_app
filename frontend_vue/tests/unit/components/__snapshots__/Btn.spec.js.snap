// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn.vue Matchs Snapshot 1`] = `<button>mock label</button>`;
